Automattic/wp-calypso
# Patterns: option to search in all categories

Related to https://github.com/Automattic/dotcom-forge/issues/6372

## Proposed Changes
Show the "Search in all categories" button if a user performs a search inside a category that doesn't yield results.

## Testing Instructions
1) Open `http://calypso.localhost:3000/patterns/events`
2) Search for "cta"
3) Assert that you see "0 patterns" and "Search in all categories" button
4) Click on "Search in all categories"
5) Assert that you see 3 results and none of the category is selected

<img width="1209" alt="Screen Shot 2024-04-02 at 3 12 19 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/76cad969-f2a2-4c33-a52e-f74d80bfdb34">
